src/app/Library/CrudPanel/Traits/Create.php

```diff
@@ -115,19 +115,20 @@ public function syncPivot($model, $data)
 
                 // if a JSON was passed instead of an array, turn it into an array
                 if (is_string($values)) {
-                    $values = json_decode($values);
+                    $values = json_decode($values, true);
                 }
 
                 $relation_data = [];
-                foreach ($values as $pivot_id) {
-                    $pivot_data = [];
-
-                    if (isset($field['pivotFields'])) {
-                        foreach ($field['pivotFields'] as $pivot_field_name) {
-                            $pivot_data[$pivot_field_name] = $data[$pivot_field_name][$pivot_id];
-                        }
+                if (isset($field['fields'])) {
+                    foreach ($values as $pivot_row) {
+                        $relation_data[$pivot_row[$field['name']]] = Arr::except($pivot_row, $field['name']);
```

Comment:
`pivotFields()` was an undocumented feature, so I suspect very few people use it. But it's still a public feature, so we should have an upgrade step in the docs telling people "_if you've been using it, this is how it's different_".

Please help me out with a difference between the two implementations:
- in 4.1 `pivotFields` accepted an array like so... example;
- in 4.2 `pivotFields` accepts an array like so... example;

That way it's clear to people what has changed.